styx-scheduler-service/src/main/java/com/spotify/styx/WorkflowResourceDecorator.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * -\-\-
+ * Spotify Styx Scheduler Service
+ * --
+ * Copyright (C) 2017 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+package com.spotify.styx;
+
+import com.spotify.styx.model.WorkflowConfiguration;
+import com.spotify.styx.model.WorkflowInstance;
+import java.util.Set;
+
+public interface WorkflowResourceDecorator {
```

Comment:
Do you think we should do some refactoring for the package? Some of the classes can be moved to `workflow`, e.g. this one, WorkflowCache, InMemWorkflowCache.

Maybe a separated PR.